companion/src/eeprominterface.cpp

```diff
@@ -260,9 +255,7 @@ RawSourceRange RawSource::getRange(const ModelData * model, const GeneralSetting
         result.unit = sensor.unitString();
       }
       else {
-        if (singleprec) {
-          result.offset = -DBL_MAX;
-        }
+        result.offset = -DBL_MAX;
 
         switch (index) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of switch cases from 50 to at most 30. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1479)